pkg/engine/handlers/validation/validate_cel.go

```diff
@@ -79,6 +79,11 @@ func (h validateCELHandler) Process(
 	// extract CEL expressions used in validations and audit annotations
 	variables := rule.Validation.CEL.Variables
 	validations := rule.Validation.CEL.Expressions
+	for i := range validations {
```

Comment:
Wondering if this is the right place to do this check. Why can this not be done when we anyway iterate through the validations when processing them? 

I feel like this has a small (maybe neglectable ) performance impact because we do an operation which might not be needed, so early on in processing.

Reply:
I amn't sure if this is a good user experience because user will find out that `validate.cel.expressions.message` is set although he used the `validate.message` field when he wrote the policy.

Reply:
One solution is to mark the field `cel.expressions.message` as required and display a warning that `validate.message` will be ignored in case of `validate.cel` subrules.

Reply:
I don't think making any `message` field as required is a good thing.

Reply:
>Internally reference validate.message as the CEL struct. If each expression does not have its own message field, print what is in validate.message.
>
>For any expression which does have its own message field, print that message.

Printing `validate.message` out if `validate.cel.expressions.message` is not set makes sense to me.

>Why can this not be done when we anyway iterate through the validations when processing them?

Can we do something [here](https://github.com/kyverno/kyverno/blob/1b3186b8baf0bfebba565c3738243e192251f181/pkg/utils/cel/cel.go#L79)?

Reply:
> Can we do something [here](https://github.com/kyverno/kyverno/blob/1b3186b8baf0bfebba565c3738243e192251f181/pkg/utils/cel/cel.go#L79)?

No. This function convert `v1alpha1.Validation` to `cel.ExpressionAccessor` so that it can be compiled. The `v1alpha1.Validation.Message` needs to be set before calling this function.